extras/foreman_protector/dnf-foreman-protector.py

```diff
@@ -0,0 +1,86 @@
+import dnf
+import dnf.exceptions
+from dnfpluginscore import _, logger
+import urlgrabber
+import urlgrabber.grabber
+import os
+import fnmatch
+import tempfile
+import time
+
+import configparser
+
+class ForemanProtector(dnf.Plugin):
+    name = 'dnf-foreman-protector'
+
+    def __init__(self,base,cli):
+        self.base = base
+        self.cli = cli
+
+    def _config(self):
```

Comment:
The hook is called `config`, not `_config`. Therefore currently this code is not executed.

Reply:
You can also utilize read_config() method of the plugin:
`parser = self.read_config(self.base.conf)`